plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -271,4 +287,90 @@ public static HoodieTableFileSystemView getFileSystemView(
     }
+
+    public static HiveHudiPartitionInfo buildHiveHudiPartitionInfo(HudiTableHandle tableHandle, String partitionName, Partition partition)
+    {
+        return new HiveHudiPartitionInfo(
+                tableHandle.getSchemaTableName(),
+                Location.of(tableHandle.getBasePath()),
+                partitionName,
+                partition,
+                tableHandle.getPartitionColumns(),
+                tableHandle.getPartitionPredicates());
+    }
+
+    public static Partition buildPartition(String partitionPath, List<Column> partitionColumns, HudiTableHandle tableHandle, PartitionValueExtractor partitionValueExtractor)
+    {
+        if (partitionPath == null || partitionPath.isEmpty()) {
+            return Partition.builder()
+                    .setDatabaseName(tableHandle.getSchemaName())
+                    .setTableName(tableHandle.getTableName())
+                    .withStorage(storageBuilder ->
+                            storageBuilder.setLocation(tableHandle.getBasePath())
+                                    .setStorageFormat(StorageFormat.NULL_STORAGE_FORMAT))
+                    .setColumns(ImmutableList.of())
+                    .setValues(ImmutableList.of())
+                    .build();
+        }
+        else {
+            List<String> values = partitionValueExtractor.extractPartitionValuesInPath(partitionPath);
+
+            if (partitionColumns.size() != values.size()) {
+                throw new TrinoException(HUDI_INVALID_PARTITION_VALUE, "Invalid partition path: " + partitionPath);
+            }
+
+            return Partition.builder()
+                    .setDatabaseName(tableHandle.getSchemaName())
+                    .setTableName(tableHandle.getTableName())
+                    .withStorage(storageBuilder ->
+                            storageBuilder.setLocation(getFullPath(tableHandle.getBasePath(), partitionPath))
+                                    // ToDo - set valid storage format
+                                    .setStorageFormat(StorageFormat.NULL_STORAGE_FORMAT))
+                    .setColumns(partitionColumns)
+                    .setValues(values)
+                    .build();
+        }
+    }
+
+    private static String getFullPath(String basePath, String relativePartitionPath)
+    {
+        return basePath.endsWith(DELIMITER_STR)
+                ? basePath + relativePartitionPath
+                : basePath + DELIMITER_STR + relativePartitionPath;
+    }
+
+    public static String getHivePartitionName(Partition partition)
+    {
+        List<Column> columns = partition.getColumns();
+        List<String> values = partition.getValues();
+
+        if (columns.size() != values.size()) {
+            throw new TrinoException(HUDI_INVALID_PARTITION_VALUE, "mismatch between column names and values, keys: " + columns.size() + " values: " + values.size());
```

Comment:
```suggestion
            throw new TrinoException(HUDI_INVALID_PARTITION_VALUE, "The number of column names and values do not match. Number of column names: " + columns.size() + ", number of values: " + values.size());
```

Reply:
Should we guarantee that the passed-in `partition` has matching `columns` and `values` and remove this check?